File: pset6/credit/credit.py
from cs50 import get_int, get_string

# list variables used
count = 0
evens = 0
odds = 0
dig1 = 0
dig2 = 0
first2 = 0

# get credit card number
num = get_int("number: ")

# calculate card value
while int(num) > 0:
    dig2 = int(dig1)
    dig1 = int(num % 10)
    if int(count) % 2 == 0:
        odds += int(dig1)
    else:
        evens = int(int(evens) + (int(dig1) * 2 / 10) + (int(dig1) * 2 % 10))
    count += 1
    num = int(num / 10)

first2 = int(dig1) * 10 + int(dig2)

# check if card is valid
if (evens + odds) % 10 == 0:
    valid = True
else:
    valid = False

# return corresponding card type
if dig1 == 4 and count >= 13 and count <= 16 and valid:
    print("VISA")
elif first2 >= 51 and first2 <= 55 and count == 16 and valid:
    print("MASTERCARD")
elif (first2 == 34 and valid) or (first2 == 37 and valid):
    print("AMEX")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int count = 0;
    int evens = 0;
    int odds = 0;
    int dig1 = 0;
    int dig2 = 0;
    int first2 = 0;
    bool valid;
    
    long num = get_long("number: ");
    
    while (num>0)
    {
        dig2 = dig1;
        dig1 = num%10;
        if (count%2 == 0)
        {
            odds+=dig1;
        }
        else
        {
            evens = evens + (dig1*2/10) + (dig1*2%10);
        }
        count++;
        num/=10;
    }
    
    first2 = dig1*10 + dig2;
    
    valid = (evens+odds)%10 == 0;
    
    if (dig1==4 && count>=13 && count<=16 && valid)
    {
        printf("VISA\n");
    }
    else if (first2>=51 && first2<=55 && valid && count==16)
    {
        printf("MASTERCARD\n");
    }
    else if ((first2==34 && valid) || (first2==37 && valid))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
